test/test_ff_utils.py

```diff
@@ -864,20 +926,38 @@ def mocked_search(url, auth, ff_env, retry_fxn):
 @pytest.mark.unit
 @pytest.mark.parametrize('url', ['', 'to_become_full_url'])
 def test_search_metadata_unit(integrated_ff, url):
+    """
+    Test normal case of search_metadata involving a search always being based on a consistently indexed set of data.
+    """
     with mock.patch.object(ff_utils, "authorized_request", make_mocked_search()):
         check_search_metadata(integrated_ff, url)
 
 
+@pytest.mark.unit
+@pytest.mark.parametrize('url', ['', 'to_become_full_url'])
+def test_search_metadata_inserting_unit(integrated_ff, url):
+    """
+    Tests unusual case of search_metadata (C4-336) returning duplicated items.
+    See detailed explanation in ``InsertingMockedSearchItems``.
+    """
+    InsertingMockedSearchItems.reset()
+    with mock.patch.object(ff_utils, "authorized_request",
+                           make_mocked_search(item_maker=InsertingMockedSearchItems.handler)):
+        check_search_metadata(integrated_ff, url, expect_shortfall=True)
+
+
 @pytest.mark.integratedx
 @pytest.mark.flaky
 @pytest.mark.parametrize('url', ['', 'to_become_full_url'])
 def test_search_metadata_integrated(integrated_ff, url):
     check_search_metadata(integrated_ff, url)
 
 
-def check_search_metadata(integrated_ff, url):
+def check_search_metadata(integrated_ff, url, expect_shortfall=False):
     if url != '':  # replace stub with actual url from integrated_ff
         url = integrated_ff['ff_key']['server'] + '/'
+
+    InsertingMockedSearchItems.reset()
```

Comment:
It's not clear to me that this function call does anything in this test. There are no mocks active here, you are just triggering (repeatedly) a side-effect that is never realized? Am I missing something here?

Reply:
Ah, I needed to expand to see that there is an instance where the mock is active - but in the case of the integrated test it is dead code.

Reply:
It's a bit obscure. I added a comment.